_scripts/customJS/tokenParser.js

```diff
@@ -0,0 +1,575 @@
+
+        switch (token.token) {
+            // date options - value is expected to be a normalizedDate object //
+            // to do: actually check this //
+            case "startdate":
+                if (pageDateInfo.startDate) value = pageDateInfo.startDate
+                formatter = "date"
+                break;
+            case "targetdate":
+                value = targetDate ? targetDate : DateManager.getTargetDateForPage(metadata)
+                formatter = "date"
+                break;
+            case "enddate":
+                if (pageDateInfo.endDate) value = pageDateInfo.endDate
+                formatter = "date"
+                break;
+            case "lastknowndate":
+                let wb = WhereaboutsManager.getWhereabouts(metadata, targetDate)
+                if (wb.lastKnown.location && wb.lastKnown.awayEnd.sort <= targetDate.sort) {
+                    value = wb.lastKnown.awayEnd
+                } else {
+                    if (targetDate) {
+                        value = targetDate
+                    } else {
+                        value = DateManager.getTargetDateForPage(metadata)
+                    }
+                }
+                formatter = "date"
+                break;
+            // end date options //
+
+            // start name options - can be a string or a name object //
+            case "name":
+                value = metadata.name ? metadata.name : file.name
+                formatter = "name"
+                break;
+            case "ancestry":
+                value = metadata.ancestry
+                formatter = "name"
+                break
+            case "subtypeof":
+                value = metadata.subTypeOf
+                formatter = "name"
+                break
+            case "typeof":
+                // special case where we define a specific main type based on metadata //
+                value = metadata.species ? metadata.species : this.#getDefaultTypeOf(metadata)
+                formatter = "name"
+                break;
+            case "subspecies":
+                value = metadata.subspecies
+                formatter = "name"
+                break
+            case "species":
+                value = metadata.species
+                formatter = "name"
+                break
+            case "person":
+                value = metadata.person
+                formatter = "name"
+                break
+            // end name options //
+
+            // start none options //
+            case "pronouns":
+                if (metadata.pronouns) value = metadata.pronouns
+                else if (metadata.gender == "male") value = "he/him"
+                else if (metadata.gender == "female") value = "she/her"
+                else if (metadata.gender) value = "they/them"
+                formatter = "none"
+                break;
+            case "age":
+                if (pageDateInfo.age == 0) value = "0 years"
```

Comment:
Certainly not something that needs to be done in this pr but it would probably be better to have an age and population formatter rather than applying the format rules here.

Reply:
Added age formatter in [80d4245](https://github.com/tsackton/taelgar/pull/25/commits/80d4245b81db5b9d48f71e1df1713bb044bd835e); not bothering with population formatter yet.